### PR TITLE
docs: consume corporate-action mutations durably

### PR DESCRIPTION
**Pull request by 0xgleb**
## Stack Context

This is the design slice for the automated corporate-actions stack. It replaces the deprecated announcement-polling source introduced in [#240](https://github.com/ST0x-Technology/st0x.issuance/pull/240); the implementation follows in [#282](https://github.com/ST0x-Technology/st0x.issuance/pull/282).

## Motivation

Advances [RAI-1043](https://linear.app/makeitrain/issue/RAI-1043/source-issuances-freeze-schedule-from-alpaca-corporate-actions-rai). Alpaca corporate actions can be inserted, revised, or deleted after a freeze window has been armed. Polling a bounded snapshot cannot prove deletions or reliably rediscover every revised future ex-date, so stale jobs can freeze an underlying on the wrong date or leave it frozen after cancellation.

## Solution

Record the decision to consume Alpaca's replayable corporate-actions SSE mutation stream and project its latest revision durably. The ADR defines:

- the exact accepted US cash/stock-dividend mutation contract and typed poison boundaries
- first-install history plus inclusive `since_id` replay with an exact anchor check
- durable cursor-regression, poison, and replay-gap boundaries that survive restart
- stable action-ID ownership for scheduled jobs and freeze holds
- transactional projection intent followed by idempotent, revision-serialized alignment
- bounded alignment retries with one deduplicated durable operator alert per terminal key
- explicit active-window update, deletion, and retention-repair semantics

## Checks

By submitting this for review, I'm confirming I've done the following:

- [x] made the design self-contained, including bootstrap, replay, poison, and retention behavior
- [x] made this PR as small as possible
- [x] linked the relevant Linear issue and implementation PR
- [x] implementation and regression coverage follow in #282

## Summary by CodeRabbit

* **New Features**
  * Added durable ingestion of corporate-action updates through an authenticated event stream.
  * Supports validated dividend event creation, updates, and deletions with replay protection.
  * Preserves scheduling and hold ownership while applying replacements, cancellations, and releases.
  * Adds startup recovery, cursor tracking, and fail-closed repair procedures for missed events.

* **Documentation**
  * Documented event validation, ordering, retention-gap handling, monitoring, and recovery workflows.
  * Removed the deprecated periodic announcements synchronization approach.

### CODE REVIEW

**Comment by 0xgleb** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/281?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#282** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/282?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#236](https://github.com/ST0x-Technology/st0x.issuance/pull/236) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/236?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#243](https://github.com/ST0x-Technology/st0x.issuance/pull/243) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/243?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#281** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/281?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/281?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#256** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/256?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#244** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/244?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#240** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/240?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#255** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/255?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#306** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/306?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#332** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/332?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2 other dependent PRs ([#293](https://github.com/ST0x-Technology/st0x.issuance/pull/293) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/293?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#335](https://github.com/ST0x-Technology/st0x.issuance/pull/335) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/335?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#331** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/331?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#317** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/317?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#324** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/324?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#309** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/309?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#308** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/308?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#303** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/303?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#302** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/302?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#301** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/301?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#300** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/300?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#298** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/298?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#296** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/296?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#289** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/289?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#288** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/288?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#287** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/287?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#286** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/286?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#285** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/285?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#280** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/280?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#279** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/279?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#277** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/277?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#275** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/275?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#274** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/274?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#273** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/273?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#271** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/271?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#268** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/268?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#267** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/267?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#216** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.issuance/216?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>add-to-gt-merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/ST0x-Technology?ref=merge-queue-instructions-in-stack-comment&prId=281).

<h2></h2>
<sub>

An organization admin has required the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: adrs/02-consume-corporate-action-mutations.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+last applied Alpaca event ID locally and reconnect with replay enabled. Treat
+the replay as at-least-once delivery and deduplicate by event ID.
+
+Project the stream into a local corporate-action schedule keyed by Alpaca's
+stable corporate-action ID. That ID, rather than ex-date-derived window
+boundaries, owns the scheduled jobs and `FreezeHoldId`. Applying an insert or
+update reconciles the source-owned window to the latest symbol and ex-date;
+applying a delete removes its pending transitions and releases its active hold.
+Persist the mutation result and replay cursor in one SQLite transaction so a
+crash can only replay work, never skip it.
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply adrs/02-consume-corporate-action-mutations.md]: Addressed across this stack with a deliberate durable two-phase boundary rather than incorrectly claiming that SQLx projection writes, Apalis job rows, and CQRS aggregate hold events share one transaction. This ADR now specifies that the accepted mutation, desired schedule revision (`reconciled_event_id = NULL`), and replay cursor commit atomically. Idempotent reconciliation then enqueues the exact revision and marks it reconciled only after enqueue succeeds; startup reconciles all pending revisions before reconnecting, and reconciliation failure stops the feed. Upstack PR #282 implements this contract and covers the crash boundaries in `projection_transaction_rolls_back_cursor_when_schedule_write_fails`, `reconciliation_enqueues_and_marks_the_projected_revision`, and `reconciliation_recovers_after_enqueue_before_marker`.

**Review by rouzwelt — APPROVED** [review]

**Review by JuaniRios — APPROVED** [review]

File: SPEC.md [hunk 2]
```diff
@@ -1611,18 +1611,42 @@ with `unfreeze_at` still ahead (window in progress) freezes immediately. This is
...
+reason, and startup refuses to reconnect while that blocked boundary remains.
+The first install requests history with `since=1970-01-01T00:00:00Z`. Subsequent
+connections use inclusive `since_id=<committed-event-id>` replay, and the first
+data frame must echo that cursor. A rejected anchor or non-echoing first frame
+is a replay gap: no later or live event is accepted.
+
+Repairing a replay-retention gap is an explicit fail-closed operation, never a
+cursor reset. While minting remains gated, the repair buffers a newly connected
+live stream durably, exhausts a paginated current-action GET snapshot for the
+accepted US dividend types, rebuilds source-owned schedules and holds, applies
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply SPEC.md]: Addressed in 9f986b81: the repair GET contract now names exactly cash_dividend,stock_dividend, while the SSE contract separately names the two corporateaction_event discriminators. The live endpoint already uses only the SSE discriminators.

File: SPEC.md [hunk 3]
```diff
@@ -1611,18 +1611,42 @@ with `unfreeze_at` still ahead (window in progress) freezes immediately. This is
...
-a corporate-action window's hold.
+**Corporate-actions sourcing.** Issuance consumes Alpaca's corporate-actions SSE
+stream as an authenticated, durable mutation source. The accepted contract is
+exactly US-region `cash_dividend_corporateaction_event` and
+`stock_dividend_corporateaction_event` frames with `insert`, `update`, or
+`delete`; any other discriminator, region, malformed identity, or invalid date
+is a typed poison boundary and cannot advance the cursor. Each action's stable
+Alpaca ID owns one source hold and one current schedule revision, so updates
+replace that action's prior window and deletes release only that action's hold.
+An operator hold or another action on the same underlying is never affected.
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply SPEC.md]: Addressed across 9f986b81 and 85c800c0: the contract now defines an unlisted valid US dividend as a cursor-only no-op. The feed implementation and regression test verify no mutation projection, schedule, alignment job, or hold is created.

File: SPEC.md [hunk 4]
```diff
@@ -1611,19 +1611,42 @@ with `unfreeze_at` still ahead (window in progress) freezes immediately. This is
...
-already elapsed are skipped; a failed pass is retried at the next interval, and
-because re-arming the same window is an idempotent no-op the sync needs no state
-of its own. The issuer CLI owns an independent operator hold and cannot release
-a corporate-action window's hold.
+**Corporate-actions sourcing.** Issuance consumes Alpaca's corporate-actions SSE
+stream as an authenticated, durable mutation source. The accepted contract is
+exactly US-region `cash_dividend_corporateaction_event` and
+`stock_dividend_corporateaction_event` frames with `insert`, `update`, or
+`delete`; any other discriminator, region, malformed identity, or invalid date
+is a typed poison boundary and cannot advance the cursor. Each action's stable
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply SPEC.md]: Addressed in 9f986b81 and verified by the implementation tests: only insert, update, and delete are accepted; unsupported kinds poison without cursor advancement. The decoder enforces an exact 64 KiB frame limit and CorporateActionId enforces 1 through 128 bytes, with boundary and oversized-input tests.

File: SPEC.md [hunk 5]
```diff
@@ -1611,19 +1611,42 @@ with `unfreeze_at` still ahead (window in progress) freezes immediately. This is
...
+The projection transaction persists the mutation, action revision, pending
+alignment marker, and replay cursor together. Apalis jobs and aggregate hold
+events are recoverable second-phase effects; startup aligns every pending
+revision before reconnecting.
+
+Event IDs are canonical uppercase ULIDs ordered by their encoded value. An exact
+replay is a duplicate; an unseen lower ID is a cursor regression. Before a
+poison or regression stops the feed, issuance persists its event ID and typed
+reason, and startup refuses to reconnect while that blocked boundary remains.
+The first install requests history with `since=1970-01-01T00:00:00Z`. Subsequent
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply SPEC.md]: Addressed across 9f986b81 and 85c800c0: SPEC and the ADR now require snapshot plus durable buffering for the first baseline. The implementation no longer requests since=1970 and has a regression test proving a missing baseline fails closed with BaselineRequired.

File: SPEC.md [hunk 6]
```diff
@@ -1611,19 +1611,42 @@ with `unfreeze_at` still ahead (window in progress) freezes immediately. This is
...
+is a replay gap: no later or live event is accepted.
+
+Repairing a replay-retention gap is an explicit fail-closed operation, never a
+cursor reset. While minting remains gated, the repair buffers a newly connected
+live stream durably, exhausts a paginated current-action GET snapshot for the
+accepted US dividend types, rebuilds source-owned schedules and holds, applies
+the buffered mutations in order, and atomically establishes the resulting
+cursor. Normal consumption resumes only after pending alignment completes;
+incomplete pagination, buffer overflow, disconnect, or an invalid row aborts
+repair.
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply SPEC.md]: Addressed in 9f986b81: the repair snapshot now has inclusive process_date start 1970-01-01 and end equal to the repair-start UTC date, and absence is authoritative only inside that accepted scope. The repair operation is not exposed in this slice, so there is no executable deletion path to test yet; the runbook keeps the feed stopped rather than allowing an unsafe partial repair.

File: SPEC.md [hunk 7]
```diff
@@ -1611,19 +1611,42 @@ with `unfreeze_at` still ahead (window in progress) freezes immediately. This is
...
+connections use inclusive `since_id=<committed-event-id>` replay, and the first
+data frame must echo that cursor. A rejected anchor or non-echoing first frame
+is a replay gap: no later or live event is accepted.
+
+Repairing a replay-retention gap is an explicit fail-closed operation, never a
+cursor reset. While minting remains gated, the repair buffers a newly connected
+live stream durably, exhausts a paginated current-action GET snapshot for the
+accepted US dividend types, rebuilds source-owned schedules and holds, applies
+the buffered mutations in order, and atomically establishes the resulting
+cursor. Normal consumption resumes only after pending alignment completes;
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply SPEC.md]: Addressed across 9f986b81 and 85c800c0: the contract explicitly commits an empty rebuilt projection without a cursor, keeps that stream open, and restarts repair if it disconnects before the first accepted frame. The current slice cannot manufacture an undefined anchor because normal consumption now rejects a missing cursor with BaselineRequired.

File: SPEC.md [hunk 8]
```diff
@@ -1611,19 +1611,42 @@ with `unfreeze_at` still ahead (window in progress) freezes immediately. This is
...
+revision before reconnecting.
+
+Event IDs are canonical uppercase ULIDs ordered by their encoded value. An exact
+replay is a duplicate; an unseen lower ID is a cursor regression. Before a
+poison or regression stops the feed, issuance persists its event ID and typed
+reason, and startup refuses to reconnect while that blocked boundary remains.
+The first install requests history with `since=1970-01-01T00:00:00Z`. Subsequent
+connections use inclusive `since_id=<committed-event-id>` replay, and the first
+data frame must echo that cursor. A rejected anchor or non-echoing first frame
+is a replay gap: no later or live event is accepted.
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply SPEC.md]: Addressed across 9f986b81 and 85c800c0: first install is specified as snapshot plus durable buffering with minting and normal consumption gated until cutover. Because this slice does not yet expose the repair operation, it now fails closed instead of treating unavailable older mutations as a complete replay; the missing-baseline regression test covers that boundary.

File: SPEC.md [hunk 9]
```diff
@@ -1614,19 +1614,71 @@ with `unfreeze_at` still ahead (window in progress) freezes immediately. This is
...
+of complete history. Minting and normal consumption remain gated until the
+rebuilt projection and cursor cutover commit. Subsequent connections use
+inclusive `since_id=<committed-event-id>` replay, and the first data frame must
+echo that cursor. `Last-Event-Id` is never sent, because Alpaca gives that
+header precedence over `since_id`. A rejected anchor or non-echoing first frame
+is a replay gap: no later or live event is accepted.
+
+Repairing a replay-retention gap is an explicit fail-closed operation, never a
+cursor reset. While minting remains gated, the repair buffers a newly connected
+live stream opened without `since`, `since_id`, or `Last-Event-Id`, then
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply SPEC.md]: Addressed: the repair buffer is now capped at both 1,024 complete frames and 64 MiB. Both limits are checked before each append; the overflowing frame and later frames are rejected, repair aborts, and issuance stays gated. Boundary coverage during active pagination is specified.

File: adrs/02-consume-corporate-action-mutations.md [hunk 10]
```diff
@@ -0,0 +1,254 @@
...
+  `cash_dividend_corporateaction_event` and
+  `stock_dividend_corporateaction_event`; the complete mutation allowlist is
+  `insert`, `update`, and `delete`. A new discriminator, another region, or an
+  unsupported mutation is a typed poison event, never an ignored frame. Action
+  IDs, symbols, ex-dates, and uppercase ULID event IDs are parsed before use. A
+  valid accepted event for an unlisted underlying is the one non-poison
+  boundary: it advances the cursor as an explicit no-op without projecting the
+  action or creating a schedule, transition, or hold.
+- Repudiation: persist every accepted event ID with the resulting schedule
+  revision so the upstream mutation and local gate state remain attributable.
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply adrs/02-consume-corporate-action-mutations.md]: Addressed: the ADR now persists a typed processing outcome for every accepted event. Listed events record schedule_revision with the revision; unlisted events record no_op_unlisted_underlying. The outcome and cursor advance are in the same projection transaction, while the unlisted path still creates no projection, schedule, transition, or hold.

File: SPEC.md [hunk 11]
```diff
@@ -1614,19 +1614,71 @@ with `unfreeze_at` still ahead (window in progress) freezes immediately. This is
...
+projection, schedule, transition job, or hold. Each listed action's stable
+Alpaca ID owns one source hold and one current schedule revision, so updates
+replace that action's prior window and deletes release only that action's hold.
+An operator hold or another action on the same underlying is never affected.
+
+The streaming decoder bounds untrusted input before JSON decoding: one SSE frame
+is at most 64 KiB, with only the separator bytes buffered beyond that limit, and
+an Alpaca action ID is 1 through 128 bytes. An oversized frame is a typed poison
+boundary before JSON or action allocation and cannot advance the cursor. Empty
+or oversized action IDs are likewise typed poison boundaries.
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply SPEC.md]: Addressed: the SPEC and ADR now persist a poison boundary with the last accepted cursor, optional canonical provider event ID, typed reason, and fixed 32-byte SHA-256 frame fingerprint. The fingerprint is incremental for oversized frames, and restart coverage is required for malformed and oversized frames without IDs.

File: adrs/02-consume-corporate-action-mutations.md [hunk 12]
```diff
@@ -0,0 +1,254 @@
...
+Consume Alpaca's corporate-actions SSE feed as a durable mutation stream,
+requesting only the two US dividend event types enumerated above. Persist the
+last applied Alpaca event ID locally and reconnect with replay enabled. Treat
+the replay as at-least-once delivery and deduplicate by event ID.
+
+Event IDs are canonical uppercase ULIDs and their encoded lexicographic order is
+the cursor order. Repeating the current cursor is an idempotent duplicate; a
+previously unseen lower ID is a poison regression, not a reason to move the
+cursor backward. Before terminating, the service durably records the typed
+blocked reason and the event ID when the rejected frame exposes one. A poison
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply adrs/02-consume-corporate-action-mutations.md]: Addressed: inclusive replay now looks up the accepted-mutation row by event_id before projection writes. Canonically identical duplicates leave the mutation, schedule, revision, and cursor unchanged; the same ID with different canonical content persists a typed poison boundary in the projection transaction.

File: adrs/02-consume-corporate-action-mutations.md [hunk 13]
```diff
@@ -0,0 +1,278 @@
...
+- Spoofing and tampering: authenticate only to Alpaca's documented stream and
+  require `region = "us"`. The complete accepted event-type allowlist is
+  `cash_dividend_corporateaction_event` and
+  `stock_dividend_corporateaction_event`; the complete mutation allowlist is
+  `insert`, `update`, and `delete`. A new discriminator, another region, or an
+  unsupported mutation is a typed poison event, never an ignored frame. Action
+  IDs, symbols, ex-dates, and uppercase ULID event IDs are parsed before use. A
+  valid accepted event for an unlisted underlying is the one non-poison
+  boundary: it advances the cursor as an explicit no-op without projecting the
+  action or creating a schedule, transition, or hold.
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply adrs/02-consume-corporate-action-mutations.md]: Addressed in both the ADR and SPEC. Unlisted accepted mutations now retain their canonical payload; after a listing commits, a service-owned reactor promotes the latest non-delete mutation per action into a pending revision without moving the SSE cursor or rewriting the historical no-op outcome. The normative tests cover both future and already-active windows.

File: adrs/02-consume-corporate-action-mutations.md [hunk 14]
```diff
@@ -0,0 +1,278 @@
...
+outcome, and cursor writes either all commit or all roll back. Exact duplicate
+events follow the unchanged-state path above. Apalis transition rows, aggregate
+hold events, and the final reconciled marker are deliberately not in that
+transaction: they are recoverable second-phase effects. The second phase
+enqueues idempotent alignment jobs for the pending revision and marks it
+reconciled only after enqueueing succeeds. Startup reconciles every pending
+revision to completion — successful enqueue and reconciled marker for each —
+before connecting to the stream. A reconciliation failure keeps the feed
+disconnected and preserves the pending revision; the feed never opens over an
+unreconciled revision.
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply adrs/02-consume-corporate-action-mutations.md]: Addressed by defining the durable alignment states from the revision marker plus keyed apalis row: pending, terminal_failed, and reconciled. Terminal failure leaves the marker null; startup queues one deduplicated failure notification and resets the same killed/exhausted job to pending with a fresh budget. Restart coverage is now required for both terminal forms.

File: adrs/02-consume-corporate-action-mutations.md [hunk 15]
```diff
@@ -0,0 +1,278 @@
...
+each complete frame, repair checks that neither ceiling would be exceeded.
+Hitting either ceiling aborts repair before that frame is appended, keeps
+issuance gated, and accepts no further frames from that stream. The repair
+rebuilds the current action projection and source-owned holds from the snapshot,
+then applies every buffered mutation in order. The snapshot itself carries no
+SSE cursor, so the cutover cursor is the last buffered mutation's event ID; the
+rebuilt projection and that cursor become authoritative in one SQLite
+transaction. If the buffer is empty when the snapshot is exhausted, the
+transaction commits the projection without a cursor and the still-open stream
+continues: the first accepted live frame commits as the new cursor, and a
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply adrs/02-consume-corporate-action-mutations.md]: Addressed consistently in the ADR and SPEC. Repair now captures the last complete buffered SSE ID as cutover boundary B before the snapshot, requires the snapshot contract to prove completeness through B, applies only buffered mutations newer than B in encoded order, and atomically commits the rebuilt projection plus the resulting cursor.

File: adrs/02-consume-corporate-action-mutations.md [hunk 16]
```diff
@@ -0,0 +1,278 @@
...
+`types` value is exactly `cash_dividend,stock_dividend`; the longer
+`*_corporateaction_event` values are SSE discriminators only. Its inclusive
+`process_date` range is `start=1970-01-01` through
+`end=<repair-start UTC date>`. The endpoint defaults an omitted range to the
+current day, so omission is forbidden; a rejected range or response that cannot
+prove complete pagination across the declared scope aborts repair. The endpoint
+has no region parameter; the US boundary comes from the US endpoint itself plus
+the accepted event types. The snapshot is authoritative for deletions only
+within that declared complete scope: an action absent from a complete scoped
+snapshot is deleted, while anything outside the scope keeps the buffered SSE
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply adrs/02-consume-corporate-action-mutations.md]: Addressed by requiring data_quality=all on every snapshot page and a provider completeness watermark/consistency guarantee before absence can be destructive. Eligibility is explicit (validated symbol, accepted type, parseable ex-date, non-delete payload); SSE-known actions outside that proven set are retained and are never deleted solely because a paginated snapshot omitted them.

File: adrs/02-consume-corporate-action-mutations.md [hunk 17]
```diff
@@ -0,0 +1,300 @@
...
+  schedule, transition, or hold, but retains the canonical mutation payload for
+  listing-time recovery. After that underlying is listed, a service-owned
+  listing reactor promotes the latest retained mutation for each action into a
+  pending schedule revision without moving the stream cursor. A retained delete
+  remains an attributable no-op because no source-owned hold was ever acquired.
+- Repudiation: persist every accepted event ID with a typed processing outcome:
+  either `schedule_revision` with the resulting revision or
+  `no_op_unlisted_underlying`. The outcome and cursor advance share the
+  projection transaction, so even an event that intentionally creates no local
+  gate state remains attributable.
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply adrs/02-consume-corporate-action-mutations.md]: Addressed in the latest #281 revision. Unlisted state is now one canonical upsert per `(region, underlying, action_id)`, with a separate payload-free audit record. Production requires positive audit-TTL and global capacity settings; cleanup is permitted only after audit expiry and after promotion/release state is no longer needed. If capacity is reached before safe cleanup, the feed blocks instead of evicting state. The same invariant is now in the SPEC.

File: adrs/02-consume-corporate-action-mutations.md [hunk 18]
```diff
@@ -0,0 +1,300 @@
...
+boundary and accept no later frame or live traffic.
+
+A retention gap cannot be repaired by silently choosing a newer cursor. The
+explicit repair procedure keeps minting gated and opens the live stream into a
+durable buffer capped at both 1,024 frames and 64 MiB of frame bytes. Before the
+first snapshot request, repair captures the last complete buffered SSE event ID
+as cutover boundary `B`; if no frame has arrived, it waits rather than inventing
+an anchor. It then fetches and exhausts a paginated GET snapshot that the
+provider contract certifies complete through `B`. A response without that
+watermark or an equivalent provider consistency guarantee aborts repair. Its
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply adrs/02-consume-corporate-action-mutations.md]: Addressed by redesigning the cutover rather than naming a capability Alpaca does not document. The ADR and SPEC now state that paginated GET cannot be related to an SSE event ID, so it is diagnostic-only and cannot rebuild state, authorize absence, or advance/reset the cursor. Production first install and retention gaps stay gated until a full issuance backup with a still-replayable cursor is restored, or Alpaca provides a documented atomic snapshot/replay boundary.

File: adrs/02-consume-corporate-action-mutations.md [hunk 19]
```diff
@@ -0,0 +1,300 @@
...
+REST `types` value is exactly `cash_dividend,stock_dividend`; the longer
+`*_corporateaction_event` values are SSE discriminators only. Its inclusive
+`process_date` range is `start=1970-01-01` through
+`end=<repair-start UTC date>`. The endpoint defaults an omitted range to the
+current day, so omission is forbidden; a rejected range or response that cannot
+prove complete pagination across the declared scope aborts repair. Every page
+requests `data_quality=all`; rejection of that request aborts repair. The
+endpoint has no region parameter; the US boundary comes from the US endpoint
+itself plus the accepted event types. Absence is authoritative only for a
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply adrs/02-consume-corporate-action-mutations.md]: Addressed in the latest #281 revision. Every diagnostic GET request now requires `region=us`, `data_quality=all`, the exact dividend type allowlist, and complete pagination. A decoded row with a missing or non-US region rejects the diagnostic result. GET remains diagnostic-only and cannot mutate the projection, release a hold, or move the SSE cursor.

File: SPEC.md [hunk 20]
```diff
@@ -1614,19 +1614,108 @@ with `unfreeze_at` still ahead (window in progress) freezes immediately. This is
...
+accepted.
+
+Repairing a replay-retention gap is an explicit fail-closed operation, never a
+cursor reset. While minting remains gated, the repair buffers a newly connected
+live stream opened without `since`, `since_id`, or `Last-Event-Id`, capped at
+both 1,024 complete frames and 64 MiB of frame bytes. Before requesting the
+snapshot, it captures the last complete buffered event ID as cutover boundary
+`B`; it waits if no complete frame exists. It then exhausts a paginated Market
+Data GET snapshot whose response contract certifies completeness through `B`. A
+missing watermark or equivalent provider consistency guarantee aborts repair.
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply SPEC.md]: Addressed by removing the unsupported buffered-snapshot repair path from both the ADR and SPEC. An empty buffer or first live frame is no longer treated as a cutover proof; production remains gated at first install or a replay gap until a replayable full backup is restored or the provider exposes a documented atomic snapshot/replay boundary. This eliminates the indefinite wait and the unsafe invented cursor.

File: SPEC.md [hunk 21]
```diff
@@ -1614,19 +1614,100 @@ with `unfreeze_at` still ahead (window in progress) freezes immediately. This is
...
+
+Unlisted canonical state is bounded: one latest row is upserted per
+`(region, underlying, action_id)`, while a separate audit row keeps only the
+event ID, action ID, mutation kind, typed outcome, payload fingerprint, and
+acceptance time. Production requires positive audit-TTL and global
+unlisted-action-capacity settings. Expired audit rows may be removed; canonical
+rows may be removed only after listing promotion reconciles, or after a
+delete/elapsed window has no schedule or hold. Reaching capacity before any row
+is eligible blocks the feed rather than silently evicting promotion or release
+state. The current replay cursor is retained independently of audit compaction.
```
[the remark this answers is not in the document]
Comment (0xgleb) [reply SPEC.md]: Addressed: the specification now couples the replay cursor to a single durable cursor-echo record containing the cursor event ID and canonical-content SHA-256 fingerprint. It defines verification after audit compaction and fails closed with a typed replay-evidence-unavailable boundary when an older duplicate lacks retained evidence. `deno fmt` passes.
